### PR TITLE
Create default pages in store locale language

### LINKED ISSUE #34420: Default pages titles created in Admin's language instead of site's language
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When creating the WC default pages from the Status > Tools menu, it creates the pages with page-titles in the language that was set for the administrator.
That is wrong as it should be in the site language.

The language setting for the administrator is just for the convenience of the administrator who is used to work with US-English back-end menu’s.

The site language is the most important one (in my case Spanish), because that determines what the visitors look at.

A work-around is to delete the English default pages, set the language for the Admin User to Spanish and then re-generate the pages within the Tools menu.

### Expected behavior

Create WC default pages from Status > Tools menu.
As you are creating these for the front-end of the website, the page titels are to be expected in the site language.
This will only happen if the language set for the administrator is same as site language.

### Actual behavior

When the back-end (!) language for the administrator (only) is set to a different language (e.g.English-US) than the site language (e.g. Spanish), the pages get titles like "shop" "my account" and "cart" instead of "tienda", "mi cuenta" and "carrito".

### Steps to reproduce

Set site language to Spanish.
Have language for admin same as site language.
Create WC default pages from Status > Tools menu.
Page titles are in Spanish.

Now delete the default pages.
Keep site language to Spanish.
Set language for admin to English-US
Create WC default pages from Status > Tools menu.
Page titles are in English instead of site language.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://staging.agya.es
Site address (URL): https://staging.agya.es
WC Version: 6.8.0
REST API Version: ✔ 6.8.0
WC Blocks Version: ✔ 8.0.0
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: es_ES
External object cache: –

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.4.30
PHP Post Max Size: 256 MB
PHP Time Limit: 300
PHP Max Input Vars: 5000
cURL Version: 7.82.0
NSS/3.67

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.35-MariaDB-cll-lve
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ❌ wp_remote_get() failed. Contact your hosting provider. 503

### Database ###

WC Database Version: 6.8.0
WC Database Prefix: Gatos_
Total Database Size: 10.95MB
Database Data Size: 8.97MB
Database Index Size: 1.98MB
Gatos_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
Gatos_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
Gatos_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
Gatos_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_aiowps_debug_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_failed_logins: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_global_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_login_activity: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_login_lockdown: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_aiowps_permanent_block: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cleantalk_ac_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cleantalk_sessions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cleantalk_sfw: Data: 0.50MB + Index: 0.42MB + Engine InnoDB
Gatos_cleantalk_sfw_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cleantalk_spamscan_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cleantalk_ua_bl: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_cli_cookie_scan: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cli_cookie_scan_categories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_cli_cookie_scan_cookies: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_cli_cookie_scan_url: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_cli_scripts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
Gatos_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_options: Data: 7.02MB + Index: 0.08MB + Engine InnoDB
Gatos_postmeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
Gatos_posts: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
Gatos_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
Gatos_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
Gatos_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
Gatos_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
Gatos_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
Gatos_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
Gatos_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
Gatos_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
Gatos_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

astra_adv_header: 1
attachment: 5
cookielawinfo: 6
custom_css: 1
customize_changeset: 47
nav_menu_item: 1
page: 7
post: 2
product: 2
revision: 5
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (19) ###

ManageWP - Worker: by GoDaddy – 4.9.14
AIOWPS Country Blocking Addon: by Peter Petreski
WPSolutions-HQ
Site Scanners
– 3.0.6

All-in-One WP Migration FTP Extension: by ServMask – 2.71
All-in-One WP Migration: by ServMask – 7.63
All In One WP Security: by All In One WP Security & Firewall Team – 4.4.12
Astra Pro: by Brainstorm Force – 3.9.1
Plantillas de inicio: by Brainstorm Force – 3.1.14
Booster Plus for WooCommerce: by Pluggabl LLC – 5.6.0
Anti-Spam by CleanTalk: by СleanTalk  – 5.183
GDPR Cookie Consent: by WebToffee – 2.1.2
Custom Fonts: by Brainstorm Force – 1.3.6
Customizer Search: by Brainstorm Force – 1.1.6
Database Cleaner & Optimizer (2022) (Pro): by Jordy Meow – 0.5.8
Plausible Analytics: by PlausibleHQ – 1.2.4
Simple Local Avatars: by Jake Goldman
10up – 2.5.0

Slim SEO: by eLightUp – 3.15.1
Spectra: by Brainstorm Force – 2.0.3
WooCommerce: by Automattic – 6.8.1
WP Bottom Menu: by J4 – 2.0.1

### Must Use Plugins (1) ###

ManageWP - Worker Loader: by GoDaddy – 1.0.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #36 - /shop/
Cart: #57 - /carrito/
Checkout: #58 - /finalizar-compra/
My account: #59 - /mi-cuenta/
Terms and conditions: ❌ Page not set

### Theme ###

Name: AGYA
Version: 1.0.0
Author URL: http://wpastra.com/about/
Child Theme: ✔
Parent Theme Name: Astra
Parent Theme Version: 3.9.1
Parent Theme Author URL: https://wpastra.com/about/?utm_source=theme_preview&utm_medium=author_link&utm_campaign=astra_theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-08-23 13:11:24 +02:00
Options: ✔
Notes: 47
Onboarding: completed

### Action Scheduler ###

Complete: 20
Oldest: 2022-08-18 16:05:27 +0200
Newest: 2022-08-22 17:40:12 +0200

Pending: 1
Oldest: 2022-08-23 17:40:12 +0200
Newest: 2022-08-23 17:40:12 +0200

### Status report information ###

Generated at: 2022-08-22 17:40:18 +02:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by kshojib**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #34420 .

This PR fixes the store locale language issues when creating default pages.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set site language to Spanish.
2. Delete the default pages (if there are any).
3. Set language for admin user to English-US.
4. Create WC default pages from Status > Tools menu.
5. Run `wp post list --post_type=page --fields=post_title,post_name`
6. Page titles and names should be created in store locale language (Spanish).

### CODE REVIEW

**Review by chihsuan — DISMISSED** [review]
Thanks for working on this! @kshojib Tested well and looks good to me. 🚢 

If you are interested, welcome to join our Slack channel for Woocommerce Contributor Day!

https://developer.woocommerce.com/2023/03/31/woocommerce-contributor-day/

**Comment by chihsuan** [comment]
Oh, looks like the PHP test CI is failing with WP <= 6.0.

https://github.com/woocommerce/woocommerce/actions/runs/4735579697/jobs/8406937168?pr=37795
https://github.com/woocommerce/woocommerce/actions/runs/4735579697/jobs/8406937310?pr=37795

**Comment by chihsuan** [comment]
I think the error occurred because the global variable [`$wp_locale_switcher`](https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/l10n.php#L1674) has not yet been defined or initialized when we call the `WC_Install::install()`. Let's see if @barryhughes has any suggestions.

**Comment by barryhughes** [comment]
> I think the error occurred because the global variable [$wp_locale_switcher](https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/l10n.php#L1674) has not yet been defined or initialized when we call the WC_Install::install()

Yes, exactly—and this is really only a problem during test suite setup (because our bootstrap code runs before `wp-settings.php`, which is where the locale switcher is instantiated). So, I think we'd need to change two further functions.

In the case of [`wp_switch_to_site_locale()`](https://github.com/woocommerce/woocommerce/blob/7.6.0/plugins/woocommerce/includes/wc-core-functions.php#L2196-L2211) let's add some extra safety (note the extra check for `$wp_locale_switcher`):

```php
/**
 * Switch WooCommerce to site language.
 *
 * @since 3.1.0
 */
function wc_switch_to_site_locale() {
	global $wp_locale_switcher;

	if ( function_exists( 'switch_to_locale' ) && isset( $wp_locale_switcher ) ) {
		switch_to_locale( get_locale() );

		// Filter on plugin_locale so load_plugin_textdomain loads the correct locale.
		add_filter( 'plugin_locale', 'get_locale' );

		// Init WC locale.
		WC()->load_plugin_textdomain();
	}
}
```

We would also need to do effectively the same thing for [`wc_restore_locale()`](https://github.com/woocommerce/woocommerce/blob/7.6.0/plugins/woocommerce/includes/wc-core-functions.php#L2213-L2228). @kshojib does that all make sense and are you happy to add another commit covering those things?

**Comment by kshojib** [comment]
Hey @chihsuan Thanks for addressing the error. @barryhughes I have added another [commit](https://github.com/woocommerce/woocommerce/pull/37795/commits/ea85b1843cd97f8a8c446f75a8a382f79363398b) with your suggestions.

File: plugins/woocommerce/changelog/fix-34420 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Create default pages in store locale language
```
Comment (barryhughes) [inline plugins/woocommerce/changelog/fix-34420]:
```suggestion
When creating default storefront pages, the site language (and not the language of the current user) should be used.
```

**Review by barryhughes — APPROVED** [review]
Perfect, thank you for addressing that! Just waiting on our automated checks passing, and then we can go ahead and merge :+1:

**Pull request merged**